### ISSUE TITLE
META GitHub Bot

### ISSUE THREAD

**Issue by moul** [gnolang/gno#3238]
Do not hesitate to let me know if individual issues could be clearer, but I believe it's better to avoid the noise.

**Comment by moul** [comment]
So, here, I clicked on "the pull request description", then the bot switched back to unchecked. Should be persisted.

![CleanShot 2024-11-28 at 17 57 33](https://github.com/user-attachments/assets/840d9a8e-9b93-4832-8832-0a90bf436de8)

**Comment by moul** [comment]
Please change the logo; it is generally poor and not suitable for small sizes.

**Comment by moul** [comment]
![CleanShot 2024-11-28 at 17 59 39](https://github.com/user-attachments/assets/37b0e3a1-c3ad-49d5-b6f1-96dd7806088b)

The introduction should be aimed at those who are not familiar with the bot. It could say something like, "I'm a bot that assists the core team in maintaining this repository. <details> followed by the details."

**Comment by moul** [comment]
![CleanShot 2024-11-28 at 18 00 59](https://github.com/user-attachments/assets/66c80877-7f0f-4108-ac87-dbb7b9a8569d)

Group the two "details" into a single "debug" at the end.

**Comment by aeddi** [comment]
> So, here, I clicked on "the pull request description", then the bot switched back to unchecked. Should be persisted.

Morgan already pinged me about this. The bot unchecks because it can't fetch the list of members from the core-contributors team, so it can't verify if you're allowed to check. It might be a token issue, I'll look into it today.

**Comment by aeddi** [comment]
Another issue: https://github.com/gnolang/gno/actions/runs/12082335167/job/33693214789?pr=3096
```
Error: unable to update pull request: unable to create status on PR 3096: POST https://api.github.com/repos/gnolang/gno/statuses/2516635949b9e0b19e9b6a4c04115cf6fbf877c4: 403 Resource not accessible by personal access token []
```

**Comment by moul** [comment]
`🔴 The pull request head branch must be up-to-date with its base`

This is already blocked by GitHub settings. 

You can keep the check if you believe the bot explains it better, but it shouldn't block the merge. It should be marked "orange" to indicate a warning.

**Comment by aeddi** [comment]
> Please change the logo; it is generally poor and not suitable for small sizes.

![Logo Short On White Background](https://github.com/user-attachments/assets/8784ea89-d954-421e-adc3-7856f48a4b92)

What about this one?

**Comment by thehowl** [comment]
> This is already blocked by GitHub settings.

Actually, not anymore; I don't recall exactly why, but this was disabled some short while ago.

**Comment by aeddi** [comment]
I always have this message in the checks of my PR:
<img width="844" alt="Screenshot 2024-12-02 at 14 33 54" src="https://github.com/user-attachments/assets/f6cbe033-733e-4dcc-b931-0b616efc5c05">

Do you think it’s not blocking / that it doesn’t put the CI in "failure" state if everything else is green?

**Comment by jefft0** [comment]
I agree that the "out-of-date with the base branch" message is unnecessary. In addition, it can lead the author of the PR to keep clicking the "Update branch" button, which we don't want. (See [this message](https://github.com/gnolang/gno/pull/2333#issuecomment-2343808931) from Leon to a user who was doing this.)

**Comment by thehowl** [comment]
>I always have this message in the checks of my PR:

yeah, it's a warning, but not a requirement. maybe we can enable it back since Teritori is now submitting PRs which are modifiable by us, and we have a check to ensure that the branch is modifiable by maintainers.

**Comment by moul** [comment]
https://github.com/gnolang/gno/pull/3135 -> green, but red.

![CleanShot 2024-12-02 at 23 18 53](https://github.com/user-attachments/assets/e435377b-b7ec-44ff-97ce-bde4c77252e6)
![CleanShot 2024-12-02 at 23 18 58](https://github.com/user-attachments/assets/3b2fb994-f546-4ea6-afb0-82ab7d96d6e5)

**Comment by jefft0** [comment]
On PR https://github.com/gnolang/gno/pull/2597, the "stale bot" added the stale label and a comment. This triggered the meta guidelines bot to add a comment. Is this considered "activity" on the PR and does it reset the timer for the stale bot?

![image](https://github.com/user-attachments/assets/de878b97-f4a5-4bae-b16c-7f83f5250e83)

**Comment by aeddi** [comment]
> #3135 -> green, but red.

Still the same error as [here](https://github.com/gnolang/gno/actions/runs/12082335167/job/33693214789?pr=3096), basically, the bot is unable to create a commit status like this:

<img width="831" alt="Screenshot 2024-12-02 at 15 57 04" src="https://github.com/user-attachments/assets/ad7fa90a-5f63-4809-bdf3-8ff8f7e4e932">
<img width="832" alt="Screenshot 2024-12-02 at 15 58 38" src="https://github.com/user-attachments/assets/e105888c-ff34-4d7f-8c8d-56fdf08babbb">

That's the last known issue for now. I created a test Github organisation and I'm currently trying to reproduce this issue to understand what settings needs to be modified on gnolang/gno.

**Comment by aeddi** [comment]
> On PR #2597, the "stale bot" added the stale label and a comment. This triggered the meta guidelines bot to add a comment. Is this considered "activity" on the PR and does it reset the timer for the stale bot?

Mmmh yes I think it should unfortunately reset the stale timer, the stale bot will probably remove the label when it will run tonight at 1:30am.

I think there are two workarounds:
- Run the GitHub Bot on the different open PRs to reset the timer now (rather than waiting for each of them to receive a stale label).
- Add an if statement to the GitHub Actions bot workflow to ignore the 'labeled' event when the author happens to be the stale bot.

**Comment by jefft0** [comment]
> > On PR #2597, the "stale bot" added the stale label and a comment. This triggered the meta guidelines bot to add a comment. Is this considered "activity" on the PR and does it reset the timer for the stale bot?
> 
> Mmmh yes I think it should unfortunately reset the stale timer, the stale bot will probably remove the label when it will run tonight at 1:30am.

You're right, @aeddi . The stale bot removed the stale label on PR https://github.com/gnolang/gno/pull/2597 . This will cause lots of confusion. I prefer your option "Add an if statement to the GitHub Actions bot workflow to ignore the 'labeled' event when the author happens to be the stale bot." In general, I think the bots should ignore each other's actions.

**Comment by aeddi** [comment]
Okay, so after digging a bit, I just saw that it wasn't the addition of the label by the stale bot that triggered the Gno bot. I think that the events emitted by the `GitHub Actions` account (used by the stale bot) are not forwarded to any other workflows, because even if the Gno bot is subscribed to the `pull_request: labelled` event, it's not fired by the stale bot applying a label.

However, what seems to have happened after analyzing the [PR](https://github.com/gnolang/gno/pull/2597) and this [workflow run](https://github.com/gnolang/gno/actions/runs/12131578319?pr=2597+%28comment%29) is that:
- The stale bot, via the `GitHub Actions` account, applied the `stale` label.
- This triggered Codecov, which is not an action or a workflow, but a GitHub app (which explains why it is triggered by events that are not forwarded to GitHub Actions workflows).
- Codecov edited its own comment, which triggered the workflow of the Gno bot.

So I will ignore the events emitted by Codecov in the Gno bot's workflow, but I wonder if the fact that Codecov edits its own comment every time the stale bot adds the stale label generates activity on the PR that causes the stale bot to remove the label afterwards.

That could explained why the stale bot removed the label right after adding it on another PR when the Gno bot was not running yet.

**Comment by aeddi** [comment]
FYI, the bot was just added as a contributor with a write role to the gno repo, so this last issue seems to be fixed : https://github.com/gnolang/gno/issues/3238#issuecomment-2514895884

**Comment by jefft0** [comment]
> FYI, the bot was just added as a contributor with a write role to the gno repo, so this last issue seems to be fixed : [#3238 (comment)](https://github.com/gnolang/gno/issues/3238#issuecomment-2514895884)

Yes, good news. The stale bot just added the stale label on PR https://github.com/gnolang/gno/pull/2655, the GnoD2D bot added its comment, and the stale label is still there. :-)

**Comment by aeddi** [comment]
@jefft0 Unfortunately, I think the bot will notice the new comment and remove the label during its next run tonight at 1:30am.

**Comment by jefft0** [comment]
> @jefft0 Unfortunately, I think the bot will notice the new comment and remove the label during its next run tonight at 1:30am.
> 
> I will open a PR to ignore event from codecov this morning.

Oh, sorry. You're right. I'll check again tomorrow.

**Comment by jefft0** [comment]
> > @jefft0 Unfortunately, I think the bot will notice the new comment and remove the label during its next run tonight at 1:30am.
> > I will open a PR to ignore event from codecov this morning.
> 
> Oh, sorry. You're right. I'll check again tomorrow.

As you said, the stale label was removed at 1:30am.

**Comment by aeddi** [comment]
@jefft0 I just merged the PR to ignore the events emitted by Codecov. 
Can you let me know if you see the problem occurring again? Normally, the Gno bot should no longer create comments on a PR marked as `stale`. 

Previously, the issue was:
- stale bot applies a label -> codecov updates its comment -> Gno bot creates its comment -> stale bot removes the label

Now it should be:
- stale bot applies a label -> codecov updates its comment -> stale bot removes the label

So we will still have to find a solution but codecov.

**Comment by jefft0** [comment]
Yes, I'll keep an eye on whether the stale bot triggers the Gno bot.

**Comment by moul** [comment]
![CleanShot 2024-12-06 at 20 34 05](https://github.com/user-attachments/assets/9312370a-227b-40ee-82ef-971ec32a1f79)

This PR is not on a fork (oops). The error is a false positive.
-> #2911

**Comment by jefft0** [comment]
There is a dependabot PR and the [Gno bot commented "Maintainers must be able to edit this pull request"](https://github.com/gnolang/gno/pull/3155#issuecomment-2525306268). How can this be resolved for PRs from dependabot? Should the Gno bot exclude dependabot PRs?

**Comment by aeddi** [comment]
I think there are two different points here:
- the rule "Maintainer can modify" should only apply to a PR opened from a fork, I have to add a condition that checks for this.
- We can choose to ignore PRs from Dependabot, but it might be useful to have automation or even manual checks on Dependabot PRs, so I'm not convinced that it's a good idea.

**Comment by moul** [comment]
![CleanShot 2024-12-08 at 12 39 07](https://github.com/user-attachments/assets/d23be1d2-cd2e-4fbb-bb03-b93864d25398)


Please add a `🐛 bug report` link to this issue in the debug section.

**Comment by moul** [comment]
Can you add a checkbox "SKIP, do not block the CI conditions are not met for this PR".

Also, we should write "who" clicked a checkbox.

**Comment by moul** [comment]
It seems that the `internal/config` should be in the `main.go`, nope?

**Comment by aeddi** [comment]
> Please add a 🐛 bug report link to this issue in the debug section.
> Move These requirements are defined in this [configuration file]. in the debug section too.

Ok 👍 

> Also, we should write "who" clicked a checkbox.

This is already the case.

> Can you add a checkbox "SKIP, do not block the CI conditions are not met for this PR".

What do you mean? All the other checks have preconditions to determine if they are displayed and a whitelist of user/team who can check them or not, etc. So how would this check work? When do we display it? Who is allowed to check it? etc.
If the goal is just to ignore the bot in certain specific cases, why would we complicate the code by adding an option instead of just merging the PR even if the bot's requirements are not met (like for any other CI check / requirement)? 

> It seems that the `internal/config` should be in the main.go, nope? 

Since the config is written in golang and the subcommand in `internal/check` needs to access it, it can't be in the main package. But we could chose to write the config in another format like json or yaml (that would require quite a bit of work).

**Comment by moul** [comment]
![CleanShot 2024-12-08 at 15 50 42](https://github.com/user-attachments/assets/f748df98-8967-479e-bfda-80107aa8ed8f)

I'll open a PR to remove this unused PR template, as it creates unnecessary noise in our Git history.

Please ensure that the bot includes a list of expected items in a top-level collapsible section named "Contributor Checklist," which should also contain a link to the CONTRIBUTING.md file.

Additionally, we need a manual checkbox for the reviewer to confirm that they have verified the required items are met.

**Comment by moul** [comment]
I've a new template suggestion:

```
#### 🛠 PR Checks Summary  
All **Automated Checks** passed. ✅  

##### Manual Checks (for Reviewers):  
- [ ] **SKIP**: Do not block the CI for this PR.  
- [ ] **Contribution Guidelines**: Verify that the PR matches our contribution guidelines.  
- [ ] **Additional Checks**: Add specific checks based on the context of the PR.  

---

<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.  

##### Automated Checks (for Contributors):  
- Automated checks are performed to ensure compliance with project requirements (e.g., passing tests, linting).  
- Address any automated check failures before proceeding.  

##### Manual Checks (for Reviewers):  
- Reviewers must manually verify specific criteria that cannot be automated.  

**Contributor Actions**:  
1. Fix any issues flagged by automated checks.  
2. Follow the Contributor Checklist to ensure your PR is ready for review.  

**Reviewer Actions**:  
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.  

##### Contributor Checklist:  
- [ ] Add new tests, or document why they are unnecessary.  
- [ ] Provide clear examples/screenshots, if necessary.  
- [ ] Update documentation, if required.  
- [ ] Ensure no breaking changes, or include `BREAKING CHANGE` notes.  
- [ ] Link related issues/PRs, where applicable.  

**Resources**:  
- [Report a bug with the bot](https://github.com/owner/repo/issues).  
- [View the bot’s configuration file](https://github.com/owner/repo/config).  

##### Debug

If possible, a single collapsible section, avoid too many nested levels.

</details>
```

---

Result:

#### 🛠 PR Checks Summary  
All **Automated Checks** passed. ✅  

##### Manual Checks (for Reviewers):  
- [ ] **SKIP**: Do not block the CI for this PR.  
- [ ] **Contribution Guidelines**: Verify that the PR matches our contribution guidelines.  
- [ ] **Additional Checks**: Add specific checks based on the context of the PR.  

---

<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.  

##### Automated Checks (for Contributors):  
- Automated checks are performed to ensure compliance with project requirements (e.g., passing tests, linting).  
- Address any automated check failures before proceeding.  

### Manual Checks (for Reviewers):  
- Reviewers must manually verify specific criteria that cannot be automated.  

**Contributor Actions**:  
1. Fix any issues flagged by automated checks.  
2. Follow the Contributor Checklist to ensure your PR is ready for review.  

**Reviewer Actions**:  
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.  

##### Contributor Checklist:  
- [ ] Add new tests, or document why they are unnecessary.  
- [ ] Provide clear examples/screenshots, if necessary.  
- [ ] Update documentation, if required.  
- [ ] Ensure no breaking changes, or include `BREAKING CHANGE` notes.  
- [ ] Link related issues/PRs, where applicable.  

**Resources**:  
- [Report a bug with the bot](https://github.com/owner/repo/issues).  
- [View the bot’s configuration file](https://github.com/owner/repo/config).  

##### Debug

If possible, a single collapsible section, avoid too many nested levels.

</details>

**Comment by thehowl** [comment]
@moul looks good, though the most important change likely to add onto the bot is a "Skip" feature to override all other checks. The rest seems implementable using the current manual checks system in the config :)

**Comment by leohhhn** [comment]
@aeddi 

[This PR](https://github.com/gnolang/gno/pull/3301) was made by me, and I'm in the @gnolang/devrels group; I guess the bot should not add the devrel check if a devrel made the PR.

**Comment by aeddi** [comment]
@leohhhn 

I'm not sure I understand what you mean, but the rule (which we can change if necessary) is that each change in the docs folder should either be:
- authored by a member of `devrels` and reviewed by a member of `tech staff`
- authored by a member of `tech-staff` and reviewed by a member of `devrels`

But in this case, there was an issue: gfanton review did not update the comment, I triggered the bot manually and I'm in the process of subscribing the bot to pull request review.

**Comment by thehowl** [comment]
Workflow fails on pull_request_review: https://github.com/gnolang/gno/actions/runs/12233559264/job/34120925785?pr=3195

**Comment by Kouteki** [comment]
> SKIP: Do not block the CI for this PR

Could we rephrase this? The statement is ambiguous, and I'm not sure if ticking the box:

1. skips the CI, or
2. runs the CI

Proposed copy:
`SKIP CI: this PR will not be blocked by failed CI checks`

**Comment by moul** [comment]
It’s not about skipping the CI, but about skipping the CI for the bot only.
But I agree we should rephrase.

**Comment by moul** [comment]
I think I already asked this here or on Signal, but I can’t remember the response.
What do you think about making the “up-to-date with base branch” check just a warning (or even removing it)? I really dislike seeing the CI red for this. If enforcing it isn’t a priority (I personally don’t mind skipping it), I believe we can configure GitHub to show it as “green but not mergeable” if needed.

**Comment by aeddi** [comment]
Personally, my opinion is that we should follow the KISS principle more. I think we are heading in the direction where the bot will become complex, hard to maintain for us and hard to understand for the contributors. And that at some point, everyone will just ignore it, which will make it useless.

For me, if we want to keep things healthy, we should force ourselves to adhere to these simple constraints:
- Either a rule is mandatory in all cases and for everyone
- Or a rule is only mandatory in certain cases and for certain people (blacklist or whitelist)
- Or we completely remove the rule

And we stick to green or red conclusions like any other CI check.

Warnings or skippable "rules", IMO -> just noise that make difficult the user's understanding and the readability of the truly relevant and mandatory rules.

For example, as previously discussed, it would have seemed healthier to me to add the `tech-staff` members to a whitelist for the rule 'does the infra need to be updated?' (which I also find annoying BTW) or even to completely remove it, rather than adding a global skip button.

But TBH, I don't really have a strong opinion on the matter. I'm just giving you my opinion, but I don't mind doing it differently.

**Comment by moul** [comment]
Please, remove the "up-to-date" check.

**Comment by aeddi** [comment]
@moul @Kouteki Quickly fixed both in this PR: https://github.com/gnolang/gno/pull/3357
Tried to reword the button description to ignore the bot as clearly as possible. Let me know if that works for you.

**Comment by zivkovicmilos** [comment]
> Please, remove the "up-to-date" check.

@moul 

Might want to reconsider this, I strongly advise having master merged into the branch and CI passing always, because we just merged this PR (it passed), but upon the master merge everything started going bananas:

https://github.com/gnolang/gno/pull/3315

**Comment by thehowl** [comment]
>Might want to reconsider this, I strongly advise having master merged into the branch and CI passing always, because we just merged this PR (it passed), but upon the master merge everything started going bananas

Was there not a way to have the requirement at the GitHub level?

I recon that's better, if we can have the GitHub bot work to enforce not using organization forks (which don't allow us to push to the branch).

**Comment by aeddi** [comment]
Ok note for later: `Fixes #<link-to-issue-comment>` also closes the issue.

**Comment by thehowl** [comment]
It looks like my fix didn't stop the huge amount of back-references. Alas. 😢 

I think the `reviewer` requirement probably double-counts reviews done by the same person. If, say, I made two Approval reveiws, that would count twice.

Also, I think that given the influx of issues on the bot has now more or less stabilized, we could probably close this issue and move instead towards creating individual issue for any further feature request / bug?

**Comment by aeddi** [comment]
I'm fine with closing this issue, I've been using gh in the CLI to check it out without the tons of back-references. 
We could replace the link in the bot's message that points to the issue with a link to an issue template instead? I can take care of it if you want.

**Comment by alexiscolin** [comment]
@aeddi @zivkovicmilos @moul Would it be possible for me to be added as a code owner for `pkg/gnoweb`? I’d would be great to be notified whenever new related code is pushed so I can review any front-end or design aspects, complementing @gfanton’s work. I used to be a code owner in the previous file, so it would be very helpful to continue with this kind of system.

**Comment by zivkovicmilos** [comment]
> [@aeddi](https://github.com/aeddi) [@zivkovicmilos](https://github.com/zivkovicmilos) [@moul](https://github.com/moul) Would it be possible for me to be added as a code owner for `pkg/gnoweb`? I’d would be great to be notified whenever new related code is pushed so I can review any front-end or design aspects, complementing [@gfanton](https://github.com/gfanton)’s work. I used to be a code owner in the previous file, so it would be very helpful to continue with this kind of system.

Yes, of course

**Comment by thehowl** [comment]
https://github.com/gnolang/gno/actions/runs/13680671069/job/38251994850?pr=2307

Need to add the "dismissed" review state

**Comment by thehowl** [comment]
a small problem on the rule for the gnoweb codeowners: https://github.com/gnolang/gno/pull/3890/files#r1985516122

**Comment by aeddi** [comment]
@thehowl I answered for the gnoweb codeowners, totally ok to make the change but not sure if it's the right way to do it.

About the automatic requests on tech-staff / external PRs, could you describe the problem more precisely? Or even send me a link to a PR that is not functioning as it should?

**Comment by thehowl** [comment]
@aeddi I'll let alexis/guilhem decide, but I think that if either of them makes a PR and I (or anyone else) approves it it should still be good. This time when Guilhem is OOO is a good case in point.

>About the automatic requests on tech-staff / external PRs, could you describe the problem more precisely? Or even send me a link to a PR that is not functioning as it should?

Any review team PR: https://github.com/gnolang/gno/pull/3869

Because the `ReviewByTeam` rule is set for the "review team" rule, tech-staff is automatically added for review when really, in this case, it shouldn't. (If anything there should be a review request on the review team, but it doesn't currently exist in the github org)

**Comment by thehowl** [comment]
We shouldn't request a review for a user, if it is the author of the PR: https://github.com/gnolang/gno/actions/runs/14309000280/job/40099315016?pr=4067

Also, given that the bot is stabilizing, I think we can look at improving the runtime to be slightly snappier. Ideally, we can work to either (a) cache the binary of the github bot or (b) run directly a github-bot action (IMO still retaining the code in this repo, but maybe having an external repo which pushes the image as needed).

**Comment by aeddi** [comment]
@thehowl 
> @aeddi I'll let alexis/guilhem decide, but I think that if either of them makes a PR and I (or anyone else) approves it it should still be good. This time when Guilhem is OOO is a good case in point. 

> We shouldn't request a review for a user, if it is the author of the PR: gnolang/gno/actions/runs/14309000280/job/40099315016?pr=4067

Ok, just to group the tasks, I had a call with @gfanton about gnoweb and what he and @alexiscolin want to set up is:
1. Alexis can resolve conversations and approve a PR -> in this case, the bot can't do anything, Alexis just needs [write permissions](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role) on the gno repo to do that. 
2. a. If Alexis or Guilhem is the author of a PR related to gnoweb, the other is assigned as an optional reviewer.
  b. If someone else is the author of a PR related to gnoweb, both are assigned as reviewers, but only one review from one of them is required.

And I'll take care of fixing the bug you mentioned in your last message (not assigning the author of a PR as a reviewer).

> Also, given that the bot is stabilizing, I think we can look at improving the runtime to be slightly snappier. Ideally, we can work to either (a) cache the binary of the github bot or (b) run directly a github-bot action (IMO still retaining the code in this repo, but maybe having an external repo which pushes the image as needed).

If that's okay with you, I can take care of implementing the option A. Option B seems to take longer to set up, and actions based on Docker (I assume that's what you're thinking of since you're talking about images) [are slower than regular actions](https://docs.github.com/en/actions/sharing-automations/creating-actions/about-custom-actions#docker-container-actions). 
So the difference between option A (restoring the binary from the cache) and option B (fetching the container) is probably not significant.

> One more thing, I noticed that the rule of the review team automatically requests a review from tech-staff on all external PRs.
>
> We'll likely need to add something like we did for the labels, on the ReviewBy rule as well.

Okay, so if I understand correctly, your idea is that as long as the review team hasn't pre-approved and the `review/triage-pending` label is still assigned to a PR, `tech-staff` shouldn't be assigned as reviewers. 
Do you want us to automatically request a review from the `tech-staff` team once the PR is approved by the review team? Or should we let the review team chose which user/team to assign?

**Comment by thehowl** [comment]
> And I'll take care of fixing the bug you mentioned in your last message (not assigning the author of a PR as a reviewer).

👍 

> So the difference between option A (restoring the binary from the cache) and option B (fetching the container) is probably not significant.

SGTM

> Do you want us to automatically request a review from the `tech-staff` team once the PR is approved by the review team? Or should we let the review team chose which user/team to assign?

Yes, I think it was already informally like that.

What I was referring to with "labels" is what we did here:

```go
func Label(gh *client.GitHub, name string, action LabelAction) Requirement {
	return &label{gh, name, action}
}
```

So that the LabelAction controls what to do with the label, given that the requirement could be chained with `Not` or other rules:

```go
const (
	// LabelApply will place the label on the PR if it doesn't exist.
	LabelApply = iota
	// LabelRemove will remove the label from the PR if it exists.
	LabelRemove
	// LabelIgnore always leaves the label on the PR as-is, without modifying it.
	LabelIgnore
)
```

**Comment by aeddi** [comment]
While checking if the last change went well, I see that in some cases we get this error: https://github.com/gnolang/gno/actions/runs/14466524060/job/40570078628?pr=4084#step:7:7 
I don't have time to deal with it right now, but I will take care of it as soon as possible.

**Comment by Kouteki** [comment]
Closing: this cycle of work is complete
